sdk/src/client/api/block_builder/input_selection/requirement/ed25519.rs

```diff
@@ -88,15 +48,15 @@ impl InputSelection {
             .available_inputs
             .iter()
             .enumerate()
-            .find(|(_, input)| input.output.is_basic() && self.available_has_ed25519_address(input, address).0)
+            .find(|(_, input)| input.output.is_basic() && self.available_has_ed25519_address(input, address))
         {
-            Some((index, None))
+            Some(index)
         } else {
             // Otherwise, checks if the requirement can be fulfilled by a non-basic output.
             self.available_inputs.iter().enumerate().find_map(|(index, input)| {
                 if !input.output.is_basic() {
-                    if let (true, account_transition) = self.available_has_ed25519_address(input, address) {
-                        Some((index, account_transition))
+                    if self.available_has_ed25519_address(input, address) {
+                        Some(index)
                     } else {
                         None
                     }
```

Comment:
some `.then_some(..)` magic is possible here if you want :trollface: (even combined with the outer if)

Reply:
Do you think this is more readable?

Reply:
to me ... it reads better, yeah, but obviously you don't need to do it.

Reply:
Done